odoo/orm/environments.py

```diff
@@ -865,7 +855,25 @@ def clear_dirty_field(self, field: Field) -> Collection[IdType]:
         return self.transaction.dirty.pop(field, ())
 
     def invalidate(self, spec: Collection[tuple[Field, Collection[IdType] | None]] | None = None) -> None:
-        return self.transaction.invalidate(spec)
+        """ Invalidate the cache, partially or totally depending on ``spec``.
+
+        If a field is context-dependent, invalidating it for a given record
+        actually invalidates all the values of that field on the record.  In
+        other words, the field is invalidated for the record in all
+        environments.
+
+        This operation is unsafe by default, and must be used with care.
+        Indeed, invalidating a dirty field on a record may lead to an error,
+        because doing so drops the value to be written in database.
+
+            spec = [(field, ids), (field, None), ...]
+        """
+        if spec is None:
+            self.transaction.invalidate()
+            return
+        env = next(iter(self.transaction.envs))
```

Comment:
I think that having `transaction` as the first argument of `_invalidate_cache` was cleaner than this.